kie-server-parent/kie-server-remote/kie-server-client/src/main/java/org/kie/server/client/impl/DocumentServicesClientImpl.java

```diff
@@ -64,6 +64,9 @@ public DocumentInstance getDocument(String identifier) {
             ServiceResponse<DocumentInstance> response = (ServiceResponse<DocumentInstance>) executeJmsCommand( script, DescriptorCommand.class.getName(), "BPM" ).getResponses().get(0);
 
             throwExceptionOnFailure(response);
+            if (shouldReturnWithNullResponse(response)) {
+                return null;
```

Comment:
that would break the api rather significantly. Putting optional still does not cover all cases and it requires additional step to get the data for non null cases. So I prefer to stick to this as it easier to deal with and majority of cases will be non null - only async handlers for jms will return null, other (request reply for jms and rest will return always non null value. Taking into consideration that REST is most likely used more than JMS it's better to stick to what we have at the moment.